### PR TITLE
[CM_BE_14] Handling OTP expiry in user creation and link flows

### PR DESCRIPTION
**Pull request by Apoorva-GA**

### CODE REVIEW

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/consent-manager/pullRequest?prid=5394788&bid=17684470)

File: src/main/java/in/projecteka/consentmanager/clients/ClientError.java [hunk 1]
```diff
@@ -116,8 +116,8 @@ public static ClientError consentArtefactForbidden() {
                 new ErrorRepresentation(new Error(CONSENT_ARTEFACT_FORBIDDEN, "Cannot retrieve Consent artefact")));
     }
 
-    public static ClientError otpNotFound() {
-        return new ClientError(NOT_FOUND, new ErrorRepresentation(new Error(OTP_INVALID, "Invalid OTP")));
+    public static ClientError invalidOtp() {
+        return new ClientError(BAD_REQUEST, new ErrorRepresentation(new Error(OTP_INVALID, "Invalid OTP")));
```
Comment (ganesanarun) [inline src/main/java/in/projecteka/consentmanager/clients/ClientError.java:120]: UnAuthorized is what swagger says If I'm not wrong.

Comment (Apoorva-GA) [reply src/main/java/in/projecteka/consentmanager/clients/ClientError.java:120]: In that case what should be the error representation to app? 
Should it say invalid otp for both invalid and expired scenario.

Comment (ganesanarun) [reply src/main/java/in/projecteka/consentmanager/clients/ClientError.java:120]: We have different error code each case no?

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           

Clones added
============
- src/test/java/in/projecteka/consentmanager/user/OtpServiceClientTest.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/consent-manager/pullRequest?prid=5394788&bid=17684470)

**Pull request merged**